dfir_lang/src/graph/hydroflow_graph.rs

```diff
@@ -448,10 +451,24 @@ impl DfirGraph {
         if matches!(self.node(node_id), GraphNode::Handoff { .. }) {
             return Some(Color::Hoff);
         }
-        // In-degree excluding ref-edges.
-        let inn_degree = self.node_predecessor_edges(node_id).count();
-        // Out-degree excluding ref-edges.
-        let out_degree = self.node_successor_edges(node_id).count();
+        // In-degree excluding ref-edges and loop-crossing edges.
+        let inn_degree = self
+            .node_predecessor_edges(node_id)
+            .filter(|&edge_id| {
+                // Filter out loop-crossing edges.
+                let pred_id = self.edge(edge_id).0;
+                self.node_loop(pred_id) == self.node_loop(node_id)
```

Comment:
I didn't understand why equality here corresponds to "loop-crossing" in the comment.